mlflow/pyfunc/model.py

```diff
@@ -228,17 +231,37 @@ def _save_model_with_class_artifacts_params(
     if artifacts:
         saved_artifacts_config = {}
         with TempDir() as tmp_artifacts_dir:
-            tmp_artifacts_config = {}
             saved_artifacts_dir_subpath = "artifacts"
             for artifact_name, artifact_uri in artifacts.items():
-                tmp_artifact_path = _download_artifact_from_uri(
-                    artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path()
-                )
-                tmp_artifacts_config[artifact_name] = tmp_artifact_path
-                saved_artifact_subpath = posixpath.join(
-                    saved_artifacts_dir_subpath,
-                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()),
-                )
+                if artifact_uri.startswith("hf:/"):
+                    try:
+                        from huggingface_hub import snapshot_download
+
+                        repo_id = re.sub(r"^hf:/", "", artifact_uri)
+                        snapshot_location = snapshot_download(
+                            repo_id=repo_id,
+                            cache_dir=os.path.join(
+                                path, saved_artifacts_dir_subpath, artifact_name
+                            ),
```

Comment:
Can we use the default `cache_dir`?

Reply:
This is to download the model into model artifacts directory

Reply:
What does the model artifacts directory look like?

```
- MLmodel
- ...
```

Reply:
<img width="1360" alt="image" src="https://github.com/mlflow/mlflow/assets/82044803/cee75972-e4d0-4417-92dc-9f2233b39956">

Reply:
Do we lob both snapshot cache and snapshot?